model/riscv_fdext_regs.sail

```diff
@@ -210,6 +245,67 @@ function wF_bits(i: bits(5), data: flenbits) -> unit = {
 
 overload F = {rF_bits, wF_bits, rF, wF}
 
+val handle_illegal : unit -> unit effect {escape, rreg, wreg}
+
+/* ---- Register Read/ Writes */
+
+val rX_or_rF_32 : (bool, bits(5)) -> (Retired, bits(32)) effect {escape, rreg}
+function rX_or_rF_32(zfinx_en, i) = {
+  let rval = if zfinx_en then nan_unbox(zfinx_en, rX(unsigned(i))) else nan_unbox(zfinx_en, rF(unsigned(i)));
```

Comment:
Just use X/F, then you don't need the unsigned conversion.

This would also benefit from being split into multiple lines.